frontend/app/src/components/hexagonalBoard/HexagonalBoard.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+	useState,
+	ReactNode,
+	useEffect,
+	forwardRef,
+	useImperativeHandle,
+} from "react";
+import HexagonalTile, {
+	HexagonalTileProps,
+} from "@/components/hexagonalTile/HexagonalTile.tsx";
+import "./HexagonalBoard.css";
+
+const defaultTileProps: HexagonalTileProps = {
+	fill: "",
+	isDimmed: false,
+	stroke: "gray",
+	strokeWidth: "16",
+	identifier: "",
+	textContent: "",
+};
+
+interface HexagonalBoardProps {
+	rows: number;
+	tilesPerRow: number;
+	ref?: React.Ref<{
+		updateTile: (
+			row: number,
+			col: number,
+			props: Partial<HexagonalTileProps>,
+		) => void;
+	}>;
+}
+
+const HexagonalBoard = forwardRef(function HexagonalBoard(
+	{ rows, tilesPerRow }: HexagonalBoardProps,
+	ref,
+) {
+	const [tileProps, setTileProps] = useState<Map<string, HexagonalTileProps>>(
+		new Map(),
+	);
+
+	useEffect(() => {
+		const initialTileProps = new Map<string, HexagonalTileProps>();
+		for (let i = 0; i < rows; i++) {
+			for (let j = 0; j < tilesPerRow; j++) {
+				const tileId = `tile-${i}-${j}`;
+				initialTileProps.set(tileId, {
+					fill: "",
+					isDimmed: false,
+					stroke: "gray",
+					strokeWidth: "16",
+					identifier: tileId,
+					textContent: "",
+				});
+			}
+		}
+		setTileProps(initialTileProps);
+	}, [rows, tilesPerRow]);
+
+	// Update a specific tile's properties
+	const updateTile = (
+		row: number,
+		col: number,
+		props: Partial<HexagonalTileProps>,
+	) => {
+		const tileId = `tile-${row}-${col}`;
+		setTileProps((prev) => {
+			const updated = new Map(prev);
+			const existingProps = updated.get(tileId) || {
+				fill: "",
+				isDimmed: false,
+				stroke: "gray",
+				strokeWidth: "16",
+				identifier: tileId,
+				textContent: "",
+			};
+			updated.set(tileId, { ...existingProps, ...props });
+			return updated;
+		});
+	};
+
+	// Expose `updateTile` via ref
+	useImperativeHandle(ref, () => ({
+		updateTile,
+	}));
+
+	const gridCellsMatrix: ReactNode[] = [];
+	for (let i = 0; i < rows; i++) {
+		const hexTiles = [];
+		for (let j = 0; j < tilesPerRow; j++) {
+			const tileId = `tile-${i}-${j}`;
+			hexTiles.push(
+				<div className="layoutedHex" key={tileId}>
+					<HexagonalTile
+						{...{ ...defaultTileProps, ...(tileProps.get(tileId) ?? {}) }}
+					/>
+				</div>,
+			);
+		}
+		gridCellsMatrix.push(
+			<div
+				className="boardRow"
+				key={`row-${i}`}
+				style={{
+					marginLeft: i % 2 === 0 ? "0px" : "140px",
+					marginTop: i === 0 ? "0px" : "-70px",
```

Comment:
Its exactly the other way around, I apply it always except for the topmost row. As you might have seen, each row of hexagons starts slightly inside the one above.

atm I think, this effect would be difficult to achieve without diving into some absolute positioning fuckery and I would like to avoid that even more. I even think this makes perfect sense here and is only very slightly hacky, since its not used to mask something but it does exactly what is written there and is easily understood if you take a look at the result.

Reply:
whoopsies, true tho :+1: